File: languages/scala/src/main/scala/de/jplag/scala/ScalaLanguage.scala
package de.jplag.scala

import de.jplag.Token
import org.kohsuke.MetaInfServices

import java.io.File
import java.util
import scala.jdk.CollectionConverters._

@MetaInfServices
class ScalaLanguage extends de.jplag.Language {

  private final val fileExtensionsList = List(".scala", ".sc")

  override def fileExtensions: util.List[String] =
    fileExtensionsList.asJava

  override def getName: String = "Scala"

  override def getIdentifier: String = "scala"

  override def minimumTokenMatch: Int = 8

  override def parse(files: util.Set[File], normalize: Boolean): util.List[Token] =
    new Parser().parse(files.asScala.toSet).asJava
}

File: languages/scala/src/main/java/de/jplag/scala/ScalaLanguage.java
package de.jplag.scala;

import java.io.File;
import java.util.List;
import java.util.Set;

import de.jplag.Language;
import de.jplag.ParsingException;
import de.jplag.Token;

import com.google.auto.service.AutoService;

/**
 * Scala language module.
 */
@AutoService(Language.class)
public class ScalaLanguage implements Language {
    @Override
    public List<String> fileExtensions() {
        return List.of(".scala", ".sc");
    }

    @Override
    public String getName() {
        return "Scala";
    }

    @Override
    public String getIdentifier() {
        return "scala";
    }

    @Override
    public int minimumTokenMatch() {
        return 8;
    }

    @Override
    public List<Token> parse(Set<File> files, boolean normalize) throws ParsingException {
        return new ScalaParser().parse(files);
    }
}
